app/controllers/admin/merchants_controller.rb

```diff
@@ -6,8 +6,37 @@ def require_admin
     render file: "/public/404" unless current_admin?
   end
 
+  def index
+    @merchants = Merchant.all
+  end
+
+  def update
+    @merchant = Merchant.find(merchant_params[:merchant_id])
+    if merchant_params[:type] == "disable"
+      @merchant.update(:active => false)
+      @merchant.items.each do |item|
+        item.update(:active? => false)
+      end
+      flash[:success] = "#{@merchant.name}'s account has been disabled."
+    elsif
+      merchant_params[:type] == "enable"
+      @merchant.update(:active => true)
+      @merchant.items.each do |item|
+        item.update(:active? => true)
+      end
+      flash[:success] = "#{@merchant.name}'s account has been enabled."
+    end
```

Comment:
This method is good candidate for refactoring. We stopped at getting working code in the system.

Reply:
Agreed. We have a similar situation with our items controller